Assets/MixedRealityToolkit.Tests/PlayModeTests/TestFixture_02_ManipulationTests.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#if !WINDOWS_UWP
+// When the .NET scripting backend is enabled and C# projects are built
+// Unity doesn't include the the required assemblies (i.e. the ones below).
+// Given that the .NET backend is deprecated by Unity at this point it's we have
+// to work around this on our end.
+using Microsoft.MixedReality.Toolkit.UI;
+using NUnit.Framework;
+using System.Collections;
+using UnityEditor.SceneManagement;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.TestTools;
+using Microsoft.MixedReality.Toolkit;
+namespace Microsoft.MixedReality.Toolkit.Tests
+{
+    public class TestFixture_02_ManipulationTests
+    {
+        /// <summary>
+        /// Test creating adding a ManipulationHandler to GameObject programmatically.
+        /// Should be able to run scene without getting any exceptions.
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator Test01_ManipulationHandlerInstantiate()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene(true);
+
+            var playspace = MixedRealityToolkit.Instance.MixedRealityPlayspace;
+            playspace.transform.position = new Vector3(1.0f, 1.5f, -2.0f);
+            playspace.transform.LookAt(Vector3.zero);
+
+            var testLight = new GameObject("TestLight");
+            var light = testLight.AddComponent<Light>();
+            light.type = LightType.Directional;
+            testLight.transform.position = new Vector3(-2.5f, 3.0f, -1.2f);
+            testLight.transform.LookAt(Vector3.zero);
+
+            var testObject = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            testObject.transform.localScale = Vector3.one * 0.2f;
+
+            var manipHandler = testObject.AddComponent<ManipulationHandler>();
+            // Wait for two frames to make sure we don't get null pointer exception.
+            yield return null;
+            yield return null;
+
+            GameObject.Destroy(testObject);
```

Comment:
Do you know why testLight doesn't need to be destroyed? Or is it that this is also intending to test that component creation -> manipulation -> destruction's entire lifecycle works as expected?

Reply:
It does, my tests are currently adding a bunch of lights to the tests, which is not correct.